addons/mail/static/src/components/call_view/call_view.xml

```diff
@@ -3,55 +3,54 @@
-                        }"
-                        t-ref="tileContainer"
-                        t-attf-style="--height:{{callView.tileHeight}}px; --width:{{callView.tileWidth}}px;"
-                    >
-                        <t t-foreach="callView.tileParticipantCards" t-as="participantCard" t-key="'grid_tile_'+participantCard.localId">
-                            <t t-if="!callView.filterVideoGrid or (participantCard.rtcSession and participantCard.rtcSession.videoStream)">
-                                <!-- maybe filter focused partner out? -->
-                                <CallParticipantCard
-                                    className="'o_CallView_participantCard o_CallView_gridTile'"
-                                    classNameObj="{ 'w-100' : callView.hasSidebar and callView.mainParticipantCard }"
-                                    record="participantCard"
-                                />
+            <div class="o_CallView_mainView d-flex flex-grow-1 flex-column align-items-center justify-content-center">
+                <!-- Call members display -->
+                <div class="o_CallView_participantContainer d-flex justify-content-between w-100 h-100 overflow-hidden" t-on-click="callView.onClick" t-on-mousemove="callView.onMouseMove">
+                    <t t-if="callView.mainParticipantCard">
+                        <div class="o_CallView_mainParticipantContainer d-flex flex-grow-1 justify-content-center position-relative mw-100 mh-100">
+                            <CallParticipantCard className="'o_CallView_participantCard'" record="callView.mainParticipantCard"/>
+                            <t t-if="callView.showOverlay and !callView.threadView.compact">
+                                <i t-if="callView.hasSidebar" class="o_CallView_sidebarButton position-absolute fa fa-arrow-right" title="Hide sidebar" t-on-click="callView.onClickHideSidebar"/>
+                                <i t-else="" class="o_CallView_sidebarButton position-absolute fa fa-arrow-left" title="Show sidebar" t-on-click="callView.onClickShowSidebar"/>
                             </t>
-                        </t>
+                        </div>
+                    </t>
+                    <t t-elif="callView.tileParticipantCards.length > 0">
+                        <div
+                            class="o_CallView_grid d-flex align-items-center h-100 w-100 flex-wrap justify-content-center"
+                            t-ref="tileContainer"
+                            t-attf-style="--height:{{callView.tileHeight}}px; --width:{{callView.tileWidth}}px;"
+                        >
+                            <t t-foreach="callView.tileParticipantCards" t-as="participantCard" t-key="'grid_tile_'+participantCard.localId">
+                                <t t-if="!callView.filterVideoGrid or (participantCard.rtcSession and participantCard.rtcSession.videoStream)">
```

Comment:
The condition look the same as in sidebar. I thought in previous comment that the `t-if` were different, but actually not.

So it looks more like a condition that is shared among all `CallParticipantCardView`s.
The previous comment still make sense.
I don't mind duplicating the `t-if` in both compute of the `CallParticipantCardViews` fields, but would be even better if you could introduce a concept for this relatively complex boolean expression, so that code is much easier to understand.

Reply:
It looks like I already moved the filter logic to the record generation function at https://github.com/odoo/odoo/pull/82692/files#diff-541f588ce750929063400f83d5ec2c79ea6c5f2c1267c008454a1747e92325b5R199

I will just remove the condition in the template